balloc/balloc.c

```diff
@@ -3019,45 +3141,47 @@ M0_INTERNAL int m0_balloc_create(uint64_t              cid,
 				 const struct m0_fid  *fid)
 {
 	struct m0_balloc       *cb;
-	struct m0_be_btree      btree = {};
 	struct m0_be_tx         tx    = {};
 	struct m0_be_tx_credit  cred  = {};
 	int                     rc;
+	struct m0_btree_type    bt;
 
 	M0_PRE(seg != NULL);
 	M0_PRE(out != NULL);
 
 	m0_be_tx_init(&tx, 0, seg->bs_domain,
 		      grp, NULL, NULL, NULL, NULL);
 	M0_BE_ALLOC_CREDIT_PTR(cb, seg, &cred);
-	m0_be_btree_init(&btree, seg, &ge_btree_ops);
-	m0_be_btree_create_credit(&btree, 1, &cred);
-	m0_be_btree_fini(&btree);
-	m0_be_btree_init(&btree, seg, &gd_btree_ops);
-	m0_be_btree_create_credit(&btree, 1, &cred);
-	m0_be_btree_fini(&btree);
+	m0_be_tx_credit_add(&cred, &M0_BE_TX_CREDIT_PTR(cb));
+
+	bt = (struct m0_btree_type){.tt_id = M0_BT_BALLOC_GROUP_EXTENTS,
+		.ksize = M0_MEMBER_SIZE(struct m0_ext, e_start),
+		.vsize = M0_MEMBER_SIZE(struct m0_ext, e_end),
+	};
+	m0_btree_create_credit(&bt, &cred);
```

Comment:
The latest changes will accept nr as a parameter.
Kindly include them as well and make changes accordingly.

Reply:
Done.